pipeline/trusted-resources/pkg/webhook/validation_admit_test.go

```diff
@@ -0,0 +1,364 @@
+package validation
+
+import (
+	"bytes"
+	"context"
+	"encoding/base64"
+	"net/http/httptest"
+	"net/url"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/google/go-containerregistry/pkg/authn"
+	imgname "github.com/google/go-containerregistry/pkg/name"
+	"github.com/google/go-containerregistry/pkg/registry"
+	v1types "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote"
+	"github.com/sigstore/cosign/pkg/cosign"
+	sigs "github.com/sigstore/cosign/pkg/signature"
+	"github.com/sigstore/sigstore/pkg/signature"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1"
+	faketekton "github.com/tektoncd/pipeline/pkg/client/clientset/versioned/fake"
+	remotetest "github.com/tektoncd/pipeline/test"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	fakek8s "k8s.io/client-go/kubernetes/fake"
+	"knative.dev/pkg/webhook/json"
+	"knative.dev/pkg/webhook/resourcesemantics"
+)
+
+var (
+	nameSpace      = "tekton-pipelines"
+	serviceAccount = "tekton-pipelines-webhook"
+)
+
+func Test_verifyResources(t *testing.T) {
+	ctx := context.Background()
+
+	os.Setenv("SYSTEM_NAMESPACE", nameSpace)
+	os.Setenv("WEBHOOK_SERVICEACCOUNT_NAME", serviceAccount)
+
+	sa := &corev1.ServiceAccount{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: nameSpace,
+			Name:      serviceAccount,
+		},
+	}
+
+	// task for local taskRef
+	ts := v1beta1.Task{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "tekton.dev/v1beta1",
+			Kind:       "Task"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "test-task",
+			Namespace: nameSpace,
+		},
+		Spec: v1beta1.TaskSpec{
+			Steps: []v1beta1.Step{{
+				Container: corev1.Container{
+					Image: "ubuntu",
+					Name:  "echo",
+				},
+			}},
+		},
+	}
+	tsTempered := v1beta1.Task{
```

Comment:
tempered is usually not a typical term I see - can you add a comment explaining what this means?

Reply:
this is a misspell, should be "tamper"